contracts/infrastructure/dapp/ParaswapFilter.sol

```diff
@@ -171,26 +205,24 @@ contract ParaswapFilter is BaseFilter {
             return false;
```

Comment:
Shouldn't we pass the address of `uniswapProxy` in the constructor to avoid this costly call. The point of having dedicated uniswap methods is to save gas and this call will cost between 5k and 10k gas (post Berlin). It will not change frequently, anw when it does we can just update the filter.

Reply:
`uniswapProxy` is already assigned in the constructor. This is a check that the Paraswap proxy used by the call is the same as the proxy recorded at the time the filter was created. 

Given that changes of the `uniswapProxy` are now timelocked on Paraswap's side, we could ask Paraswap to increase their timelock (currently 1 hour) to e.g. 7 days so that if they get compromised we have time to get notified and remove our filter.

Reply:
Their timelock cannot be changed without a redeployment of Augustus btw, so unless we can convince them to redeploy, we might want a cronjob on our end to kill our filter as soon as the the proxy gets changed. This could be via a method callable by anyone that lets you shutdown the filter whenever there is a pending change of the uniswapProxy.

Reply:
Agreed.

Reply:
I changed it to checking a local storage variable updated via
```
    function updateIsValidUniswapProxy() external {
        isValidUniswapProxy = (uniswapProxy == IParaswap(_augustus).getUniswapProxy());
    }
```
But obviously we don't save a ton as there is still the storage read cost.